app/questionnaire/questionnaire_schema.py

```diff
@@ -0,0 +1,168 @@
+from app.validation.error_messages import error_messages
+
+
+class QuestionnaireSchema(object):
+
+    def __init__(self, questionnaire_json):
+        self.json = questionnaire_json
+        self._groups = self._get_groups()
+        self._blocks = self._get_blocks()
+        self.error_messages = self._get_error_messages()
+        self.aliases = self._get_aliases()
+
+    def _get_groups(self):
+        groups = {}
+        for group in self.json.get('groups', []):
+            groups[group.get('id')] = group
+
+        return groups
+
+    def _get_blocks(self):
+        blocks = {}
+        for group in self.json.get('groups', []):
+            for block in group['blocks']:
+                blocks[block.get('id')] = block
+
+        return blocks
+
+    def _get_block_ids(self):
+        ids = []
+        for group in self.get_groups():
+            for block in group['blocks']:
+                ids.append(block.get('id'))
+
+        return ids
+
+    def _get_error_messages(self):
+        messages = error_messages.copy()
+        if 'messages' in self.json:
+            for key, message in self.json['messages'].items():
+                messages[key] = message
+
+        return messages
+
+    def _get_aliases(self):
+        aliases = {}
+        for block in self.get_blocks():
+            for question in self.get_questions_for_block(block):
+                for answer in question['answers']:
+                    if 'alias' in answer:
+                        if answer['alias'] in aliases:
+                            raise Exception('Duplicate alias found: ' + answer['alias'])
+                        aliases[answer['alias']] = {
+                            'answer_id': answer['id'],
+                            'repeats': answer['type'] == 'Checkbox' or question['type'] == 'RepeatingAnswer',
+                        }
+
+        return aliases
+
+    def get_first_block_id_for_group(self, group_id):
+        group = self.get_group(group_id)
+        if group:
+            return group['blocks'][0]['id']
+
+    def is_first_block_id_for_group(self, group_id, block_id):
```

Comment:
This method is only used in one place, I think we should remove it and just use `get_first_block_id_for_group` (doing the comparison at the point this method is currently called in path_finder